src/ProgressOnderwijsUtils/XmlSerializableBase.cs

```diff
@@ -20,31 +23,42 @@ public static string SerializeToString(object o)
                         .GetConstructors(BindingFlags.Instance | BindingFlags.NonPublic)
                         .Single()
                         .Invoke(null)
-                ).SerializeToInst(xw, o);
+                ).SerializeToInst(xw, o, namespaces);
         }
         return writer.ToString();
     }
 
     public static XDocument SerializeToXDocument(object o)
+        => SerializeToXDocument(o, null);
+
+    public static XDocument SerializeToXDocument(object o, XmlSerializerNamespaces? namespaces)
     {
         var doc = new XDocument();
-        using (var xw = doc.CreateWriter()) {
-            ((IXmlSerializeHelper)
-                    typeof(XmlSerializerHelper<>)
-                        .MakeGenericType(o.GetType())
-                        .GetConstructors(BindingFlags.Instance | BindingFlags.NonPublic)
-                        .Single()
-                        .Invoke(null)
-                ).SerializeToInst(xw, o);
-        }
+        using var xw = doc.CreateWriter();
+        ((IXmlSerializeHelper)
+                typeof(XmlSerializerHelper<>)
+                    .MakeGenericType(o.GetType())
+                    .GetConstructors(BindingFlags.Instance | BindingFlags.NonPublic)
+                    .Single()
+                    .Invoke(null)
+            ).SerializeToInst(xw, o, namespaces);
 
         return doc;
     }
+
+    public static string ToStringWithXmlDeclaration(this XDocument xdoc, XDeclaration declaration)
+        => $"{declaration}{Environment.NewLine}{xdoc}";
+
+    public static string ToStringWithXmlDeclaration(this XDocument xdoc, XDeclaration declaration, SaveOptions options)
+        => $"{declaration}{Environment.NewLine}{xdoc.ToString(options)}";
```

Comment:
This isn't necessary; XDocument can contain a declaration (which you're now ignoring).  To serialize the embedded declaraion, use `Save` instead of `ToString`